### PR TITLE
Drtii 1678 add central fast access latest data storage

### PR DESCRIPTION
**Pull request by richbirch**
Add tables, serialisation & dao for:
 - queue slots
 - flights

Move `CrunchMinute`, `MinuteLike` & `TQM` from drt-v2
Update indices for passengers_hourly & capacity_hourly
Add queries to delete agg table data by dates to enable data retention policy to apply

### CODE REVIEW

File: jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/QueueSlotDaoTest.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+class QueueSlotDaoTest extends AnyWordSpec  with Matchers with BeforeAndAfter {
+  sequential
+
+  private val db = TestDatabase.db
+
+  import TestDatabase.profile.api._
+
+  val dao: QueueSlotDao = QueueSlotDao()
+
+  println(dao.table.schema.createStatements.mkString(";\n") + ";")
```
Comment (ngu04) [inline jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/QueueSlotDaoTest.scala:27]: println here

File: jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/FlightDaoTest.scala [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  private val db = TestDatabase.db
+
+  import TestDatabase.profile.api._
+
+  private val portCode: PortCode = PortCode("LHR")
+
+  val dao: FlightDao = FlightDao()
+
+  println(dao.table.schema.createStatements.mkString(";\n") + ";")
```
Comment (ngu04) [inline jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/FlightDaoTest.scala:29]: println here to keep or remove ?

**Review by ngu04 — APPROVED** [review]
Nice one !  This will be useful .   do know what would be size of table in a year or 2 ? and if it is worth considering partition as this will collating all ports data together in one table ?

File: jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/QueueSlotDaoTest.scala (hunk 1 above)
Comment (richbirch) [reply jvm/src/test/scala/uk/gov/homeoffice/drt/db/dao/QueueSlotDaoTest.scala]: Yeah I'll move these to a utility function :+1:

**Comment by richbirch** [comment]
> Nice one ! This will be useful . do know what would be size of table in a year or 2 ? and if it is worth considering partition as this will collating all ports data together in one table ?

For flights roughly 400k per year, and queue slots 1m. Not enough to consider the overhead of adding partitions

**Pull request merged**